policyengine_us/variables/gov/states/mi/tax/income/exemptions/mi_exemptions.py

```diff
@@ -0,0 +1,57 @@
+from policyengine_us.model_api import *
+
+
+class mi_exemptions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Michigan exemptions"
+    defined_for = StateCode.MI
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "http://legislature.mi.gov/doc.aspx?mcl-206-30",
+        "https://www.legislature.mi.gov/Publications/TaxpayerGuide.pdf",
+    )
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.mi.tax.income.exemptions
+
+        # Personal Exemptions
+        person = tax_unit("head_spouse_count", period)
+        personal_exemption = person * p.filer
+
+        # Dependent exemptions
+        dependents = tax_unit("tax_unit_dependents", period)
+        dependent_exemption = dependents * p.dependent
+
+        # Stillborn Exemptions
+        stillbirths = tax_unit("tax_unit_stillbirths", period)
+        stillborn_exemption = stillbirths * p.stillborn
+
+        # Disabled exemptions
+        disabled_people = add(tax_unit, period, ["is_disabled"])
+        disabled_exemption = disabled_people * p.disabled
+
+        # Disabled veteran exemptions
+        disabled_veterans = add(
+            tax_unit, period, ["is_fully_disabled_service_connected_veteran"]
+        )
+        disabled_veteran_exemption = disabled_veterans * p.disabled_veteran
+
+        # Dependent on other return exemptions
+        filing_status = tax_unit("filing_status", period)
+        is_dependent_on_other_return = tax_unit("dsi", period).astype(int)
+        is_dependent_exemption = (
```

Comment:
I think one last thing is to make sure that if f.e. a person is the head but also claimed as a dependent on another return, they can only get dependent_on_another_return exemption but not the main one